onnx/defs/tensor/defs.cc

```diff
@@ -10,18 +10,15 @@ OPERATOR_SCHEMA(Cast)
 The operator casts the elements of a given input tensor to a data type
 specified by the 'to' argument and returns an output tensor of the same size in
 the converted type. The 'to' argument must be one of the data types specified
-in the 'DataType' enum field in the TensorProto message. If the 'to' argument
-is not provided or is not one of the enumerated types in DataType, Caffe2
-throws an Enforce error.
+in the 'DataType' enum field in the TensorProto message.
 
 NOTE: Casting to and from strings is not supported yet.
 )DOC")
     .Attr(
           "to",
           "The data type to which the elements of the input tensor are cast."
           "Strictly must be one of the types from DataType enum in TensorProto",
-          AttributeProto::STRING,
-          OPTIONAL)
+          AttributeProto::STRING)
```

Comment:
Should be AttributeProto::INT32? as it's enum.

Reply:
Yeah it's indeed better to use INT32. However, caffe2 supports the string version, so that means there could be "valid" model with string `to` attribute. Changing this to INT32 would be a breaking change.

Reply:
@bddppq @houseroad I'd suggest we fix it if we all agree that it's better please, though it's a breaking change.